sycl/unittests/Extensions/DefaultContext.cpp

```diff
@@ -61,3 +51,29 @@ TEST(DefaultContextTest, DefaultContextCanBeDisabled) {
   ASSERT_TRUE(catchException)
       << "ext_oneapi_get_default_context did not throw and exception";
 }
+
+TEST(DefaultContextTest, DefaultContextTest) {
+  using namespace sycl::detail;
+  using namespace sycl::unittest;
+  ScopedEnvVar var(EnableDefaultContextsName, "1",
+                   SYCLConfig<SYCL_ENABLE_DEFAULT_CONTEXTS>::reset);
+
+  test_contexts_are_equal();
+}
+
+TEST(DefaultContextTest, DefaultContextCanBeDisabled) {
+  using namespace sycl::detail;
+  using namespace sycl::unittest;
+  ScopedEnvVar var(EnableDefaultContextsName, "0",
+                   SYCLConfig<SYCL_ENABLE_DEFAULT_CONTEXTS>::reset);
+
+  test_default_context_disabled();
+}
+
+TEST(DefaultContextTest, DefaultContextCanBeDisabledEnabled) {
+  sycl::detail::enable_ext_oneapi_default_context(false);
+  test_default_context_disabled();
+
+  sycl::detail::enable_ext_oneapi_default_context(true);
+  test_contexts_are_equal();
+}
```

Comment:
Please add a test when the value is changed after one/two platforms/queues/devices have been created already.